src/helper/remote_config/client.hpp

```diff
@@ -45,11 +47,17 @@ class client {
 
     static client::ptr from_settings(const service_identifier &sid,
         const remote_config::settings &settings,
-        std::vector<remote_config::product> &&products,
-        std::vector<protocol::capabilities_e> capabilities);
+        std::optional<bool> client_enable_configuration,
+        const std::shared_ptr<dds::service_config> &service_config,
+        const std::shared_ptr<dds::engine> &engine_ptr, bool rules_file_set);
 
     virtual bool poll();
     virtual bool is_remote_config_available();
+    [[nodiscard]] virtual const std::unordered_map<std::string, product>
```

Comment:
This is only used for testing, right? However are you sure you want to copy the entire map?

Reply:
Done